packages/core/core/src/utils/transform-content-types-to-models.ts

```diff
@@ -158,6 +158,13 @@ export const transformAttribute = (
         },
       };
     }
+    case 'relation': {
+      if (attribute.required) {
+        throw new Error(
+          `Required relations are not supported, please remove it from ${contentType.collectionName} schema on the attribute ${name}`
+        );
+      }
+    }
```

Comment:
If we did allow this setup in the content type schema in v4, we should notify the docs team to include this in the list of v5 breaking changes

Reply:
I recall quite a few comments by people who added the required attribute by hand so the UI would reflect it so, yes, a huge notice alerting this as a potential breaking change should be included in the release notes.